src/pages/data-entry/data-entry.jsx

```diff
@@ -1,127 +1,214 @@
-    } else {
-      this.setState({ ...this.state, ...{ wunits: unitw.target.value } });
-      this.setState({ ...this.state, ...{ errorwunits: false } });
+  addweightUnits (unitw) {
+    let WeightUnitsArray = DataFunction()
+
+    unitw.target.value.toLowerCase()
+
+    this.setState({ ...this.state, ...{ errorwunits: true } })
+    for (i = 0; i < WeightUnitsArray.length; i++) {
+      if (unitw.target.value === WeightUnitsArray[i][0]) {
+        this.setState({ ...this.state, ...{ errorwunits: false } })
+        this.setState({ ...this.state, ...{ wunits: unitw.target.value } })
+      }
     }
   }
-  addheight(h) {
-    this.setState({ ...this.state, ...{ height: h.target.value } });
+
+  addheight (h) {
+    this.setState({ ...this.state, ...{ height: h.target.value } })
   }
-  addheightUnits(unith) {
-    if (
-      unith.target.value !== "cm" &&
-      unith.target.value !== "m" &&
-      unith.target.value !== "in" &&
-      unith.target.value !== "ft"
-    ) {
-      this.setState({ ...this.state, ...{ errorhunits: true } });
-    } else {
-      this.setState({ ...this.state, ...{ hunits: unith.target.value } });
-      this.setState({ ...this.state, ...{ errorhunits: false } });
+  addheightUnits (unith) {
+    var HeightUnitsArray = DataFunction()
+
+    unith.target.value.toLowerCase()
+
+    this.setState({ ...this.state, ...{ errorhunits: true } })
+    for (i = 0; i < HeightUnitsArray.length; i++) {
+      if (unith.target.value === HeightUnitsArray[i][0]) {
+        this.setState({ ...this.state, ...{ errorhunits: false } })
+        this.setState({ ...this.state, ...{ hunits: unith.target.value } })
+      }
     }
   }
-  addage(a) {
-    this.setState({ ...this.state, ...{ age: a.target.value } });
+
+  addage (a) {
+    this.setState({ ...this.state, ...{ age: a.target.value } })
   }
-  setsex(gen) {
+  setsex (gen) {
     if (
-      gen.target.value !== "Male" &&
-      gen.target.value !== "male" &&
-      gen.target.value !== "Female" &&
-      gen.target.value !== "female" &&
-      gen.target.value !== "Other" &&
-      gen.target.value !== "other"
+      gen.target.value !== 'Male' &&
+      gen.target.value !== 'male' &&
+      gen.target.value !== 'Female' &&
+      gen.target.value !== 'female' &&
+      gen.target.value !== 'Other' &&
+      gen.target.value !== 'other'
     ) {
-      this.setState({ ...this.state, ...{ errorsex: true } });
+      this.setState({ ...this.state, ...{ errorsex: true } })
     } else {
-      this.setState({ ...this.state, ...{ sex: gen.target.value } });
-      this.setState({ ...this.state, ...{ errorsex: false } });
+      this.setState({ ...this.state, ...{ sex: gen.target.value } })
+      this.setState({ ...this.state, ...{ errorsex: false } })
     }
   }
-  enterForm() {
+  enterForm () {
+    if (this.state.errorhunits) {
+      let newunith = prompt(
+        "We do not recognize this unit. What is it's ratio to 1 ?"

```

Comment:
What does this prompt mean? Ratio to 1? I'm not sure if that makes sense with users.